### PR TITLE
Better handle GitHub bad token

### PR DESCRIPTION
**Pull request by pmiossec**
Transform a NRE in an UnauthorizedAccessException with clearer message

First step to handle https://github.com/gitextensions/gitextensions/issues/7641

### CODE REVIEW

**Comment by pmiossec** [comment]
Just review the first commit.

File: Git.hub/Client.cs [hunk 1]
```diff
@@ -99,33 +94,59 @@ public IList<Repository> getRepositories(string username)
...
             if (repo == null)
                 return null;
 
-            repo._client = client;
+            repo._client = _client;
             repo.Detailed = true;
             return repo;
         }
 
+        private T DoRequest<T>(IRestRequest request, bool shouldNotFail = false) where T : new()
```
Comment (RussKie) [inline Git.hub/Client.cs]: Please invert `shouldNotFail`. It leads to a mental overload when staring at `true` to understand what it means.
```suggestion
        private T DoRequest<T>(IRestRequest request, bool throwOnError = true) where T : new()
```

**Review by RussKie — CHANGES_REQUESTED** [review]

File: Git.hub/Client.cs [hunk 2]
```diff
@@ -112,6 +113,32 @@ public Repository getRepository(string username, string repositoryName)
             return repo;
         }
 
+        private T DoRequest<T>(RestRequest request, bool shouldNotFail = false) where T : new()
```
Comment (RussKie) [inline Git.hub/Client.cs]: Please move the private method after publics.

File: Git.hub/Client.cs [hunk 3]
```diff
@@ -172,11 +191,11 @@ public List<Repository> searchRepositories(string query)
             var request = new RestRequest("/legacy/repos/search/{query}");
             request.AddUrlSegment("query", query);
 
-            var repos = client.Get<APIv2.RepositoryListV2>(request).Data;
-            if (repos == null || repos.Repositories == null)
-                throw new Exception(string.Format("Could not search for {0}", query));
+            var repos = DoRequest<APIv2.RepositoryListV2>(request);
+            if (repos?.Repositories == null)
```
Comment (gerhardol) [inline Git.hub/Client.cs:169]: Maybe add {} to conform with GE style (but will not be the same as other files in this submodule)

**Review by gerhardol — APPROVED** [review]
OK for me

**Comment by pmiossec** [comment]
Updated.

**Review by RussKie — APPROVED** [review]

**Review by gerhardol — APPROVED** [review]

**Pull request merged**